src/twisted/conch/ssh/transport.py

```diff
@@ -638,20 +638,39 @@ def sendPacket(self, messageType, payload):
                 payload
             ) + self.outgoingCompression.flush(2)
         bs = self.currentEncryptions.encBlockSize
-        # 4 for the packet length and 1 for the padding length
-        totalSize = 5 + len(payload)
+
+        isGCM = self.currentEncryptions.outCipType.endswith(b"-gcm@openssh.com")
+
+        # 1 for the padding length
+        totalSize = 1 + len(payload)
+
+        if not isGCM:
+            totalSize += 4
```

Comment:
I do not know enough about this logic to say that it is necessarily *wrong* but it sure is confusing. You add 4 here, then *use* totalSize, then subtract for, then subtract 4 *again* when constructing the packet itself?  The case above puts GCM in the negative case, then below it's in the positive case?

I think it might be easier to follow if the `totalSize` used to compute `lenPad` were given a different name, like `paddedSize`?

Reply:
Agree... this is an ugly hack.

The code from https://github.com/twisted/twisted/pull/12186 for the ETM PR should have a better implementation for it.

Reply:
OK, will have a look over there first. This should probably be rebased on top of it: )